interfaces/apparmor/spec_test.go

```diff
@@ -605,3 +605,66 @@ func (s *specSuite) TestSetSuppressPycacheDeny(c *C) {
 	s.spec.SetSuppressPycacheDeny()
 	c.Assert(s.spec.SuppressPycacheDeny(), Equals, true)
 }
+
+var key1 = apparmor.RegisterSnippetKey("testkey1")
+var key2 = apparmor.RegisterSnippetKey("testkey2")
+
+func (s *specSuite) TestPrioritySnippets(c *C) {
+	restore := apparmor.SetSpecScope(s.spec, []string{"snap.demo.scope1"})
+
+	// Test a scope with a normal snippet and prioritized ones
+	s.spec.AddSnippet("Test snippet 1")
+	s.spec.AddPrioritizedSnippet("Prioritized snippet 1", key1, 0)
+	s.spec.AddPrioritizedSnippet("Prioritized snippet 2", key1, 0)
+	s.spec.AddPrioritizedSnippet("Prioritized snippet 3", key2, 1)
+	s.spec.AddPrioritizedSnippet("Prioritized snippet 4", key2, 2)
+	s.spec.AddPrioritizedSnippet("Prioritized snippet 5", key2, 0)
+
+	restore()
```

Comment:
Can you leave a comment that the test is just looking for lack of panics (I presume that is the intent).

Consider skipping the part after restore into a separate test function for clarity.

Reply:
Mmm... The point is that I'm setting the scope twice, and I presume that I have to call the restore functions "in reverse"... I checked it, and the `defer XXX()` statements are executed "in reverse order", as it is needed in this function, so I have modified this.